src/kbmod/search/kernels/kernels.cu

```diff
@@ -220,6 +220,16 @@ __global__ void searchFilterImages(PsiPhiArrayMeta psi_phi_meta, void *psi_phi_v
     assert(psi_phi_vect != nullptr && image_times != nullptr && trajectories != nullptr &&
            results != nullptr);
 
+    // Copy the times to faster shared memory for the block. Make sure all threads
+    // copy their time before progressing. We need to do this before pruning on
+    // (x, y) in order to correctly handle blocks at the edge of the image.
+    __shared__ double shared_times[MAX_NUM_IMAGES];
+    int time_idx = threadIdx.x + threadIdx.y * THREAD_DIM_X;
+    if (time_idx < psi_phi_meta.num_times) {
+        shared_times[time_idx] = image_times[time_idx];
+    }
+    __syncthreads();  // Block until all are done loading.
```

Comment:
Is my understanding correct, that we call `__syncthreads` here because we have to wait for `shared_times` to be completely populated by all threads?

Reply:
Yes. That is correct. Each thread in the block loads a single time stamp.